utils/manage_events.py

```diff
@@ -0,0 +1,114 @@
+# for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing, please email
+# opensource@seagate.com or cortx-questions@seagate.com.
+
+
+import logging
+import json
+from typing import Dict, List, Optional
+
+from cortx.utils.message_bus import MessageBus, MessageConsumer
+from ha.core.event_manager.event_manager import EventManager
+from ha.core.event_manager.subscribe_event import SubscribeEvent
+
+COMPONENT_ID = 'hare'
+
+class EventListener():
+    """This is listener for event manager
+
+    Typical use case of this class is as follows
+    subscribe(event list) - subscribe to events for listening, returns topic
+                            which can be used for listening
+    _listen(topic) - reading messages received on above topic
+    unsubscribe(event list) - unsubscribe as we dont want to receive these
+                              events anymore
+    get_message_type() - returns topic just in case user is not maintaining
+                         output of 'subscribe'
+    Note: 1. There is only 1 topic(message type, queue) for given component.
+          2. When component subcribes for event for the 1st time topic is
+             created and returned to user.
+          3. This topic will be used by user to listen for events.
+          4. After event is received in queue, user will read it, process it
+             and acknowledge it
+
+    """
+
+    event_manager = None
+    consumer = None
+
+    def __init__(self):
+        logging.debug('Inside EventListener')
+        self.event_manager = EventManager.get_instance()
+
+    def subscribe(self, event_list: List[SubscribeEvent]) -> str:
+        logging.info('Subscribing to events: %s', event_list)
+        kafka_topic_name = self.event_manager.subscribe(COMPONENT_ID, event_list)
+        return kafka_topic_name
+
+    def get_message_type(self) -> str:
+        '''Gets message type name mapped with component name
+        Event manager creates message type (queue/topic) when component
+        subscribes the event for the first time. User can use this message type
+        or topic for listening events. Note it is same topic which user will
+        receive after calling subscribe method'''
+
+        logging.debug('get_message_type')
+        return self.event_manager.message_type(COMPONENT_ID)
+
+    def get_events(self) -> List[str]:
+        '''Get list of events subscribed by Hare'''
+
+        logging.debug('Get list')
+        return self.event_manager.get_events(COMPONENT_ID)
+
+    def unsubscribe(self, event_list: List[str]) -> bool:
+        logging.info('Unsubscribing for events: %s', event_list)
+        return self.event_manager.unsubscribe(COMPONENT_ID, event_list)
+
+    def get_next_message(self, topic: str, time_out) -> Optional[Dict]:
```

Comment:
Let's use more specific type instead of `Dict`. 

`Dict` has no commitments on which keys it has and what type do they have. Let's create a custom type (dataclass) that will hold the parsed message. And please add `_parse()` method that translates the raw string into that well-typed class.

Reply:
Done